nutils/evaluable.py

```diff
@@ -2048,7 +2048,8 @@ def __init__(self, func:asarray, power:asarray):
     assert func.shape == power.shape
     self.func = func
     self.power = power
-    super().__init__(args=[func,power], shape=func.shape, dtype=float)
+    dtype = float if func.dtype == power.dtype == int else _jointdtype(func.dtype, power.dtype)
```

Comment:
Follows the type rule in `numeric.power`.

Reply:
Is this the same as `_jointdtype(float, func.dtype, power.dtype)`?

Reply:
For the interesting types, yes. But `(bool, int)` gives `int`. Do we still want this or we just say that `Power` require its arguments to have `dtype ≥ int`?

Reply:
Interesting, just tried it out and indeed bool^int gives int, and negative powers raise a ValueError. Likewise for int^int, so there we already deviate. So here we have to choose either to follow numpy strictly and raise an error if the power evaluates to a negative number, or to deviate from that (as we do now, apparently) and return floats regardless. Something to discuss.

Reply:
> and negative powers raise a ValueError

I think this is why `numeric.power` is implemented in the first place.